Code/Tools/AssetProcessor/Platform/Linux/native/FileWatcher/FileWatcher_linux.cpp

```diff
@@ -188,48 +191,48 @@ void FolderRootWatch::WatchFolderLoop()
             // Break out of the loop when the notify handle was closed (outside of this thread)
             break;
         }
-        else if (bytesRead > 0)
+        if (!bytesRead)
         {
-            for (size_t index=0; index<bytesRead;)
+            continue;
+        }
+        for (size_t index=0; index<bytesRead;)
+        {
+            const auto* event = reinterpret_cast<inotify_event*>(&eventBuffer[index]);
+
+            if (event->mask & (IN_CREATE | IN_DELETE | IN_MODIFY | IN_MOVE))
             {
-                struct inotify_event *event = ( struct inotify_event * ) &eventBuffer[ index ];
+                QString pathStr = QString("%1%2%3").arg(m_platformImpl->m_handleToFolderMap[event->wd], QDir::separator(), event->name);
```

Comment:
For path operations using QT, the [QDir](https://doc.qt.io/qt-5/qdir.html) should be used over QString

i.e
```suggestion
                QString pathStr = QDir(m_platformImpl->m_handleToFolderMap[event->wd]).absoluteFilePath(event->name);
```

Reply:
Fixed.